lib/Dialect/FIRRTL/Transforms/RemoveUnusedPorts.cpp

```diff
@@ -79,12 +79,43 @@ void RemoveUnusedPortsPass::removeUnusedModulePorts(
     if (port.isInput() && !arg.use_empty())
       continue;
 
+    auto portIsUnused = [&](InstanceRecord *a) -> bool {
+      auto port = a->getInstance()->getResult(arg.getArgNumber());
+      return port.getUses().empty();
+    };
+
     // Output port.
     if (port.isOutput()) {
       if (arg.use_empty()) {
         // Sometimes the connection is already removed possibly by IMCP.
         // In that case, regard the port value as an invalid value.
         outputPortConstants.push_back(None);
+      } else if (llvm::all_of(instanceGraphNode->uses(), portIsUnused)) {
+        // If the port is unused in all of its instantiations, then replace
+        // internal references to the port with whatever is driving it.
```

Comment:
Isn't enough to unconditionally replace it with a wire, regardless of uses or connects (e.g. not search for them)?

Reply:
Yes. 😂  This is actually how it is implemented below, this comment is stale.

Originally, I RAUW the port with the src. However, that doesn't work as it may break dominance. Using a wire is always safe.

Reply:
Switched to just drop a wire unconditionally. Way better.